### DISCUSSION TITLE
Small implant customization

### DISCUSSION THREAD

**Discussion by howtodo1** [BishopFox/sliver#1039] [General]
I want to make a small change to how the sliver implant finds the C2 server. It calls `C2generator()` to find the address and attempts to connect to it. How can I change the code so that it calls `C2generator` whenever the connection fails? The output of C2generator will change everytime.

**Comment by moloch--** [comment]
https://github.com/BishopFox/sliver/wiki/External-Builders are the easiest way to create customizations that are compatible with upstream servers.
